src/util/thread.h

```diff
@@ -26,6 +26,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "../threading/mutex_auto_lock.h"
 #include "porting.h"
 #include "log.h"
+#include "container.h" // MutexedQueue
```

Comment:
Wait actually, it's not even what I initially said!

The headers are just purely broken and it really needs to be there. I'll put it in a separate commit.